include/kernel.h

```diff
@@ -3074,7 +3074,7 @@ struct k_work {
 struct k_delayed_work {
 	struct k_work work;
 	struct _timeout timeout;
-	struct k_work_q *work_q;
+	struct k_work_q *work_q; /* null if not scheduled */
```

Comment:
Might be worth adding an assertion to check this on the submission path.

Reply:
There's already a check that it's either null or equal to the queue it's to be added to.  I don't think we can do more than that.